tools/rapids-configure-sccache

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# A utility script that configures sccache environment variables
+
+if [ "${CI:-false}" = "false" ]; then
+  # Configure sccache for read-only mode since no credentials
+  # are available in local builds.
+  export SCCACHE_S3_NO_CREDENTIALS=true
+  export PARALLEL_LEVEL=${PARALLEL_LEVEL:-$(nproc)}
+  export RAPIDS_BUILD_TYPE=${RAPIDS_BUILD_TYPE:-"pull-request"}
```

Comment:
These don't seem sccache-specific. @ajschmidt8 do you recall why these needed to be set specifically in the CI=false case? That would help indicate where the proper place to set these is.

Reply:
I can at least remove `RAPIDS_BUILD_TYPE`, the prompts already take care of it

Reply:
`PARALLEL_LEVEL` is set as an environment variable on our actual self-hosted runners.

So I figured we should include a value for that variable when running locally.

I'm not sure the best place to put this now that we're splitting up some of these scripts. If there's no objections, it can stay here. Alternatively we can try to find a new script for it or remove it entirely, though I'm not sure of the implications for local builds if we remove it entirely.

It's only one line, so we could also just leave it for now and then move it to the individual repositories in the future.

The `RAPIDS_BUILD_TYPE` line can probably be removed for the reason that Divye mentioned. I was just trying to provide a sane default for people who are trying to build things locally.

Reply:
I'm OK leaving `PARALLEL_LEVEL` now and moving it later. That said, since we generally use ninja (and not make) everywhere the default of nproc is the same as what ninja will do anyway.

Reply:
We can remove it later when we confirm locally that builds work as expected without `PARALLEL_LEVEL` set in the containers

Reply:
Sounds good. Resolving.